external_compressors/pickle_external.py

```diff
@@ -0,0 +1,72 @@
+"""
+A simple compressor which serializes any given input data structure
+using pickle and converts it to bitarray. 
+
+This compresor is useful when we want to send over some metadata/side-information
+in the bitstream whose size is not a big concern
+
+NOTE: pickle is extremenly wasteful, and we can do much better with custom serialization
+so this should be only used when the metadata is small.
+"""
+
+import pickle
+from typing import Any
+from core.data_encoder_decoder import DataDecoder, DataEncoder
+from utils.bitarray_utils import BitArray, bitarray_to_uint, uint_to_bitarray
+
+
+class PickleEncoder(DataEncoder):
+    def __init__(self, length_bitwidth=32):
+        self.length_bitwidth = length_bitwidth
+
+    def encode_block(self, data: Any):
+        # pickle prob dist and convert to bytes
+        pickled_bits = BitArray()
+        bytes = pickle.dumps(data)
+        pickled_bits.frombytes(bytes)
+        len_pickled = len(pickled_bits)
+
+        # encode length of pickled data
+        assert len_pickled < (1 << self.length_bitwidth)
+        length_encoding = uint_to_bitarray(len_pickled, bit_width=self.length_bitwidth)
+        return length_encoding + pickled_bits
+
+
+class PickleDecoder(DataDecoder):
+    def __init__(self, length_bitwidth=32):
+        self.length_bitwidth = length_bitwidth
+
+    def decode_block(self, bitarray: BitArray):
+        length_bitwidth = 32
```

Comment:
why set to 32 here and not use self.length_bitwidth?

Reply:
thanks! ll fix